### PR TITLE
feat: expose @vitejs/plugin-vue config

### LINKED ISSUE #214: Expose configuration for @vitejs/plugin-vue
**Is your feature request related to a problem? Please describe.**
I am looking for a way to pass some configuration to `@vitejs/plugin-vue`, specifically `template.compilerOptions.isCustomElement`). ([code](https://github.com/vuejs/vitepress/blob/c11055c5fb31ff31502400de3e5f6f05011edd1d/src/node/plugin.ts#L33)). 

**Describe the solution you'd like**
I was think that extracting these options from `.vitepress/config.js` shouldn't be too much work. Are there any other options besides `isCustomElement` that would be worth adding?

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by ChrisShank**
Fixes #214

### CODE REVIEW

**Comment by antfu** [comment]
I personally don't feel it's good to be a root-level config option as it loses some context here. Maybe we can have a config entry that forwards the options for `@vitejs/plugin-vue` instead?

**Comment by ChrisShank** [comment]
@antfu Are there any other options from `@vitejs/plugin-vue` that you think would be beneficial to expose?

**Comment by antfu** [comment]
I am thinking to have a top-level option `vueOptions`, where

```ts
 const vuePlugin = createVuePlugin({
    include: [/\.vue$/, /\.md$/],
    ...configs.vueOptions
  })
```

WDYT?

**Comment by ChrisShank** [comment]
Ya I am happy with that! I just didnt know if we wanted the expose all of the vue plugin’s options. Thanks for the feedback, Ill update the PR later today!! 😃

**Comment by ChrisShank** [comment]
@kiaking @antfu Any other feedback for this?

File: src/node/config.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { resolveAliases, APP_PATH, DEFAULT_THEME_PATH } from './alias'
 import { SiteData, HeadConfig, LocaleConfig } from '../../types/shared'
 import { MarkdownOptions } from './markdown/markdown'
 import { AliasOptions } from 'vite'
+import { Options } from '@vitejs/plugin-vue'
```
Comment (antfu) [inline src/node/config.ts:9]: We can alias this to `VuePluginOptions` to provide the context

Comment (ChrisShank) [reply src/node/config.ts:9]: Sure!

File: src/node/config.ts [hunk 2]
```diff
@@ -21,6 +22,7 @@ export interface UserConfig<ThemeConfig = any> {
   alias?: Record<string, string>
   markdown?: MarkdownOptions
   customData?: any
+  vueOptions: Options
```
Comment (kiaking) [inline src/node/config.ts:25]: I think this should be optional?

```ts
{
  ...
  vueOptions?: Options
}
```

File: src/node/plugin.ts [hunk 3]
```diff
@@ -24,14 +24,15 @@ const isPageChunk = (
...
+  { configPath, alias, markdown, site, vueOptions }: SiteConfig,
   ssr = false,
   pageToHashMap?: Record<string, string>
 ): Plugin[] {
   const markdownToVue = createMarkdownToVueRenderFn(root, markdown)
 
   const vuePlugin = createVuePlugin({
-    include: [/\.vue$/, /\.md$/]
+    include: [/\.vue$/, /\.md$/],
+    ...vueOptions
```
Comment (antfu) [inline src/node/plugin.ts:35]: should have a fallback value here

Comment (ChrisShank) [reply src/node/plugin.ts:35]: Are you referring to the case where `vueOptions` is `undefined`? It's valid to spread `undefined` so I didn't think a fallback is necessary.

Comment (antfu) [reply src/node/plugin.ts:35]: Oh what? I thought it's invalid to spread undefined... nvm 😅

**Review by antfu — APPROVED** [review]

File: src/node/plugin.ts (hunk 3 above)
Comment (ChrisShank) [reply src/node/plugin.ts:35]: Ya I think you can also spread `null` too haha! Thanks for the code review 😄

**Comment by ChrisShank** [comment]
Actually just realized that we should print a console warning to restart the dev server if `vueOptions` are updated [similar to when `base` is updated](https://github.com/vuejs/vitepress/blob/master/src/node/plugin.ts#L148).

**Pull request merged**